File: ExampleApp/app/src/main/java/com/motiv/example/Post.java
package com.motiv.example;

import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.databinding.BindingAdapter;
import com.google.gson.*;
import com.google.gson.annotations.*;
import com.google.gson.reflect.*;
import com.squareup.picasso.Picasso;
import io.realm.*;
import java.util.*;
import java.util.concurrent.*;

public class Post extends RealmObject {

    private static final Gson gson = new Gson();
    @NonNull private java.lang.String id = UUID.randomUUID().toString();

    @SerializedName("user_id")
    private java.lang.String user_id;

    @SerializedName("_links")
    private com.motiv.example.Links links;

    @SerializedName("title")
    private java.lang.String title;

    @SerializedName("body")
    private java.lang.String body;

    public java.lang.String getId() {

        return this.id;
    }

    public void setId(java.lang.String id) {
        this.id = id;
    }

    public java.lang.String getUser_id() {

        return this.user_id;
    }

    public void setUser_id(java.lang.String user_id) {
        this.user_id = user_id;
    }

    public com.motiv.example.Links getLinks() {

        return this.links;
    }

    public void setLinks(com.motiv.example.Links links) {
        this.links = links;
    }

    public java.lang.String getTitle() {

        return this.title;
    }

    public void setTitle(java.lang.String title) {
        this.title = title;
    }

    public java.lang.String getBody() {

        return this.body;
    }

    public void setBody(java.lang.String body) {
        this.body = body;
    }

    @BindingAdapter({"bind:imageUrl"})
    public static void loadImage(ImageView view, java.lang.String url) {

        Picasso.with(view.getContext()).load(url).into(view);
    }

    public static Post fromJson(String json) {
        return gson.fromJson(json, Post.class);
    }

    public static String toJson(Post object) {
        return gson.toJson(object, Post.class);
    }

    public static String toJsonArray(Post[] array) {
        return gson.toJson(array, Post[].class);
    }

    public static Post[] fromJsonArray(String array) {
        return gson.fromJson(array, Post[].class);
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/Post.kt
 
package com.motiv.example
import androidx.annotation.NonNull
import androidx.room.*
import com.google.gson.*
import com.google.gson.annotations.*
import com.google.gson.reflect.*
import java.util.*
import java.util.concurrent.*
@Entity(tableName = "post")
public class Post {

    @NonNull
    @PrimaryKey

    private var id: String = UUID.randomUUID().toString()
    @ColumnInfo(name = "resultOwnerId")@ForeignKey(entity = com.motiv.example.PostsListResponse::class, parentColumns = ["id"], childColumns = ["resultOwnerId"])
    private
    var resultOwnerId: String = ""
    @ColumnInfo(name = "linksId")@ForeignKey(entity = com.motiv.example.Links::class, parentColumns = ["id"], childColumns = ["linksId"])
    private
    var linksId: String = ""
    @ColumnInfo(name = "user_id")
    @SerializedName("user_id")
    private var user_id: String = ""
    @Ignore
    @SerializedName("_links")
    private var links: com.motiv.example.Links = com.motiv.example.Links()
    @ColumnInfo(name = "title")
    @SerializedName("title")
    private var title: String = ""
    @ColumnInfo(name = "body")
    @SerializedName("body")
    private var body: String = ""

    fun getId(): String {
        return this.id
    } fun setId(id: String) {
        this.id = id
    } fun getUser_id(): String {
        return this.user_id
    } fun setUser_id(user_id: String) {
        this.user_id = user_id
    } fun getLinks(): com.motiv.example.Links {
        return this.links
    } fun setLinks(links: com.motiv.example.Links) {
        this.links = links
    } fun getTitle(): String {
        return this.title
    } fun setTitle(title: String) {
        this.title = title
    } fun getBody(): String {
        return this.body
    } fun setBody(body: String) {
        this.body = body
    } companion object {
        val gson: Gson = Gson()
        fun fromJson(json: String): Post {
            return gson.fromJson(json, Post::class.java)
        }

        fun toJson(obj: Post): String {
            return gson.toJson(obj, Post::class.java)
        }

        fun toJsonArray(obj: Post): String {
            return gson.toJson(obj, Array<Post>::class.java)
        }

        fun fromJsonArray(json: String): Array<Post> {
            return gson.fromJson(json, Array<Post>::class.java)
        }
    } public fun getResultOwnerId(): String { return this.resultOwnerId; }; public fun setResultOwnerId(resultOwnerId: String) { this.resultOwnerId = resultOwnerId; } public fun getLinksId(): String { return this.linksId; }; public fun setLinksId(linksId: String) { this.linksId = linksId; }
}
